src/tools/compiletest/src/runtest.rs

```diff
@@ -468,7 +468,13 @@ impl<'test> TestCx<'test> {
 
         if let Some(revision) = self.revision {
             let normalized_revision = normalize_revision(revision);
-            cmd.args(&["--cfg", &normalized_revision]);
+            let cfg_arg = ["--cfg", &normalized_revision];
+            if self.props.compile_flags.windows(2).any(|args| args == cfg_arg) {
```

Comment:
Remark: IIRC, the `compile_flags` splitting is naive above (e.g. `--meow="foo bar"` -> `-meow="foo` and `bar"` but it's better than nothing here.

Reply:
The flag splitter knows how to handle `'--meow=foo bar'`, but in all other cases it will naïvely split on whitespace yeah.